src/TableView/TableView.jsx

```diff
@@ -257,16 +287,33 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
     };
 
     handleFilterData = property => {
-      const searchFields = property === undefined ? undefined : Object.keys(property);
+      const {substringSearchEnabled, metadataSubstringSearchEnabled} = this.props;
+      const searchFields = substringSearchEnabled === true &&
```

Comment:
Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
      const searchFields = substringSearchEnabled &&
```